packages/api/tests/lib/test-mmt.js

```diff
@@ -3,33 +3,89 @@
 const test = require('ava');
 const sinon = require('sinon');
 const rewire = require('rewire');
-const { CMR } = require('@cumulus/cmr-client');
-const { randomId } = require('@cumulus/common/test-utils');
+const path = require('path');
+const { randomId, readJsonFixture } = require('@cumulus/common/test-utils');
 const cloneDeep = require('lodash/cloneDeep');
 const mmt = rewire('../../lib/mmt');
 
 const insertMMTLinks = mmt.__get__('insertMMTLinks');
 const buildMMTLink = mmt.__get__('buildMMTLink');
+const parseResults = mmt.__get__('parseResults');
+const updateResponseWithMMT = mmt.__get__('updateResponseWithMMT');
 const log = mmt.__get__('log');
 
+/**
+ *  Fakes a CMR return feed where each item in the list contains an
+ *  collection_id labeled "id" that is made of the short_name and version
+ * @param {Array<Object>} list - parsed list of collection results
+ * @returns {Object} Fake CMR responses
+ */
+const cmrReturnsWithIds = async (list) => {
+  const entry = list.map((l) => ({
+    id: `${l.short_name}-${l.version}`,
```

Comment:
The cmr collection id is something like `C987654321-CUMULUS` and doesn't contain shortname and versionid.  How would a cmr collection item not have 'collection' id?

Reply:
This is just a fake function. It is used to test that whatever is in the ID is used in the MMTLink that is created from it.  I'm using the shortName and version just for convenience in the test you can see here: https://github.com/nasa/cumulus/blob/mhs/CUMULUS-2412/update-cmr-client-mmt-information/packages/api/tests/lib/test-mmt.js#L105-L115


I actually don't know if CMR can return a collection without a collection id. These are just updated tests that I had written before.  Oh, but now that I look at them, what I would expect is a return that is empty from CMR.  I will make this function return empty objects as if the collection wasn't found in CMR.  https://github.com/nasa/cumulus/blob/mhs/CUMULUS-2412/update-cmr-client-mmt-information/packages/api/tests/lib/test-mmt.js#L38-L41. Would that be better?

Reply:
changed in 8f87a62